code/__HELPERS/icons.dm

```diff
@@ -943,6 +943,20 @@ var/global/list/friendly_animal_types = list()
 		return J
 	return 0
 
+/atom/proc/cut_overlays()
+	overlays.Cut()
+	overlays += priority_overlays
+
+/atom/proc/add_overlay(image, priority = 0, paranoid = 0)
+	if(paranoid && (image in overlays))
+		return
+	if(priority)
+		priority_overlays += image
+		overlays += image
+	else
+		overlays -= priority_overlays
+		overlays += image
+		overlays += priority_overlays

```

Comment:
Don't do this, doing it this way causes three overlay list updates to get sent to clients. as well as 3 overlay list and appearance and id array churn.

var/OL = overlays.copy()
OL -= priority_overlays
OL += image
OL += priority_overlays
overlays = OL

Also, remove paranoid, make it the default.